packages/collapsible/src/LionCollapsible.js

```diff
@@ -0,0 +1,207 @@
+   * @public
+   */
+  toggle() {
+    this.__toggle();
+  }
+
+  /**
+   * Show extra content.
+   * @public
+   */
+  show() {
+    if (!this.opened) {
+      this.__setStateAndUpdate(true);
+    }
+  }
+
+  /**
+   * Hide extra content.
+   * @public
+   */
+  hide() {
+    if (this.opened) {
+      this.__setStateAndUpdate(false);
+    }
+  }
+
+  get _invokerNode() {
+    return Array.from(this.children).find(child => child.slot === 'invoker');
+  }
+
+  get _contentNode() {
+    return Array.from(this.children).find(child => child.slot === 'content');
+  }
+
+  get _contentHeight() {
+    const size = this._contentNode.getBoundingClientRect().height;
+    return `${size}px`;
+  }
+
+  _setup() {
+    const invokerNode = this._invokerNode;
+    const contentNode = this._contentNode;
+    const uid = uuid();
+    this._setupOpenCloseListeners();
+    this.__setCommonAriaAttributes(this.opened);
+    this.__setDefaultHeight(contentNode);
+    invokerNode.setAttribute('id', `collapsible-invoker-${uid}`);
+    invokerNode.setAttribute('aria-controls', `collapsible-content-${uid}`);
+    contentNode.setAttribute('aria-labelledby', `collapsible-invoker-${uid}`);
+    contentNode.setAttribute('id', `collapsible-content-${uid}`);
+  }
+
+  _setupOpenCloseListeners() {
+    this.__toggle = () => {
+      this.__setStateAndUpdate(!this.opened);
+    };
+    this._invokerNode.addEventListener('click', this.__toggle);
+  }
+
+  /**
+   * Set current state, update content node height, and dispatch event
+   * @param {Boolean} opened current state
+   * @private
+   */
+  __setStateAndUpdate(opened) {
+    this.opened = opened;
+    this.__updateContentSize(this.opened);
+    this.__dispatchEvent();
+  }
+
+  /**
+   * Toggle extra content visibility on state change.
+   * @param {Boolean} open flag whether to open or close
+   * @private
+   */
+  async __updateContentSize(open) {
+    let expectedHeight = '0px';
+    const contentNode = this._contentNode;
+    if (open) {
+      const currentHeight = this._contentHeight; // Current height before animating
+      expectedHeight = await this.__calculateHeight(contentNode);
+      contentNode.style.setProperty('max-height', currentHeight); // Set the original size i.e. 0 to animate to expected size
+      await this._showAnimation({ contentNode });
+    } else {
+      await this._hideAnimation({ contentNode });
+    }
+    contentNode.style.setProperty('max-height', expectedHeight);
+  }
+
+  // eslint-disable-next-line class-methods-use-this, no-empty-function
+  async _showAnimation() {}
+
+  // eslint-disable-next-line class-methods-use-this, no-empty-function
+  async _hideAnimation() {}
+
+  /**
+   * Calculate total content height after collapsible opens
+   * @param {Object} contentNode content node
+   * @private
+   */
+  async __calculateHeight(contentNode) {
+    contentNode.style.setProperty('max-height', '');
+    await new Promise(resolve => requestAnimationFrame(() => resolve()));
```

Comment:
Doesn't this mean that we get a content flash in order to calculate?

Reply:
You can use iron-collapse's trick to trigger a style recalculation synchronously: https://github.com/PolymerElements/iron-collapse/blob/master/iron-collapse.js#L193 we've used this for another collapsible component and it works there.

Reply:
Ah yeah true! I know it as `void this.offsetWidth;` which I used to create the action logger:

https://github.com/ing-bank/lion/blob/master/packages/helpers/sb-action-logger/src/SbActionLogger.js#L227

Has a short explanation why it works

Reply:
Yep that works as well.